pkg/deployer/deployer.go

```diff
@@ -24,8 +24,20 @@ func New() *Deployer {
 func (d *Deployer) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
 	if machine.ObjectMeta.Annotations != nil {
 		if ip, ok := machine.ObjectMeta.Annotations[constants.OpenstackIPAnnotationKey]; ok {
-			klog.Infof("Returning IP from machine annotation %s", ip)
-			return ip, nil
+			clusterProviderSpec, err := providerv1.ClusterSpecFromProviderSpec(cluster.Spec.ProviderSpec)
+			if err != nil {
```

Comment:
How do you know this error will always be because the IP isn't there?

Reply:
The idea is not to say that the error in `ClusterSpecFromProviderSpec` is because could not get ip. The idea is to concatenate the error from `ClusterSpecFromProviderSpec` with could not get IP to provider more context to the caller, which is a usual go pattern to build stacktraces.

Reply:
Oh so the returned error will get shown in a greater context by the caller is what you're saying? How do you know that's the case here? I assume because you're familiar with what is calling functions from `deployer.go` and that's the typical pattern? Thanks for the explanation

Reply:
It's just a typical go pattern. The error is returned up the call stack and more and more context is added so you get a useful error message when you print it out. If you don't do this you'll end up with error's like "error parsing yaml" without any context and you have no idea what went wrong.